agent/checks/check.go

```diff
@@ -694,17 +695,27 @@ func (c *CheckTCP) run() {
 
 // check is invoked periodically to perform the TCP check
 func (c *CheckTCP) check() {
-	conn, err := c.dialer.Dial(`tcp`, c.TCP)
```

Comment:
I think we can simplify this whole function if we use an interface for the conn:
```go
func (c *CheckTCP) check() {
	var conn io.Closer
	var err error

	if c.TLSClientConfig == nil {
		conn, err = c.dialer.Dial(`tcp`, c.TCP)
	} else {
		conn, err = tls.DialWithDialer(c.dialer, `tcp`, c.TCP, c.TLSClientConfig)
	}

	if err != nil {
		c.Logger.Warn("Check TCP connection failed",
			"check", c.CheckID.String(),
			"error", err,
		)
		c.StatusHandler.updateCheck(c.CheckID, api.HealthCritical, err.Error())
		return
	}

	conn.Close()
	c.StatusHandler.updateCheck(c.CheckID, api.HealthPassing, fmt.Sprintf("TCP connect %s: Success", c.TCP))
}
```

WDYT?

Reply:
That's a great idea, I hadn't considered that was possible.